peer/single.go

```diff
@@ -71,3 +73,19 @@ func (s *Single) Stop() error {
 func (s *Single) IsRunning() bool {
 	return true
 }
+
+// Introspect returns a ChooserStatus with a single PeerStatus.
+func (s *Single) Introspect() introspection.ChooserStatus {
```

Comment:
Doesn't this make it impossible to have a peer implemented outside the library, since the interface is in `internal`?